frontend/src/store.js

```diff
@@ -1,76 +1,86 @@
-import { createStore } from 'vuex';
+import { createStore } from "vuex";
 import { loginUser } from "./httpLayers/login.http.js";
+import { signOut } from "./httpLayers/signout.http.js";
 
 const store = createStore({
-    state() {
-        return {
-            username: null,
-            email: null,
-            accountType: null,
-            isAuth: false
-        }
+  state() {
+    return {
+      username: null,
+      email: null,
+      accountType: null,
+      isAuth: false,
+    };
+  },
+  mutations: {
+    /**
+     * It requires object argument with username, email, and accountType properties (all Strings)
+     */
+    setUser(state, payload) {
+      state.username = payload.username;
+      state.email = payload.email;
+      state.accountType = payload.accountType;
     },
-    mutations: {
-        /**
-        * This will clear all user's data and unauthorize current session
-        */
-        logOut(state) {
-            state.username = null;
-            state.email = null;
-            state.accountType = null;
-            state.isAuth = false;
-        },
-        /**
-          * It requires object argument with username, email, and accountType properties (all Strings)
-        */
-        setUser(state, payload) {
-            state.username = payload.username;
-            state.email = payload.email;
-            state.accountType = payload.accountType;
-        }
+  },
+  actions: {
+    /**
+     * This will call loginUser function which will return logging user's data
+     * and it will be saved in state.
+     */
+    async logIn(context, payload) {
+      try {
+        let user = await loginUser(payload.username, payload.password);
+        context.commit("setUser", user);
+        this.state.isAuth = true;
+        return true;
+      } catch (error) {
+        throw new Error(error.message);
+      }
     },
-    actions: {
-        /**
-        * This will call loginUser function which will return logging user's data
-        * and it will be saved in state.
-        */
-        async logIn(context, payload) {
-            try {
-                let user = await loginUser(payload.username, payload.password);
-                context.commit('setUser', user);
-                this.state.isAuth = true;
-                return true;
-            } catch(error) {
-                throw new Error(error.message);
-            }
-        }
+    /**
+     * Signs user out of the application session.
+     *
+     * Wipes the stored user data and the session cookie.
+     *
+     * @throws {Error} if wiping out the session cookie was not successful
+     */
+    async logOut(state) {
```

Comment:
Remove 'state' argument from here. If you want get access to state in action, you can do it with
```this.state.something = whatever;```

Reply:
Done